pkg/ccl/backupccl/restore_job.go

```diff
@@ -1350,21 +1335,28 @@ func (r *restoreResumer) dropDescriptors(
 		return nil
 	}
 
-	// Needed to trigger the schema change manager.
-	if err := txn.SetSystemConfigTrigger(r.execCfg.Codec.ForSystemTenant()); err != nil {
-		return err
-	}
-
 	b := txn.NewBatch()
 
 	// Collect the tables into mutable versions.
 	mutableTables := make([]*tabledesc.Mutable, len(details.TableDescs))
 	for i := range details.TableDescs {
-		mutableTables[i] = tabledesc.NewExistingMutable(*details.TableDescs[i])
+		var err error
+		mutableTables[i], err = descriptors.GetMutableTableVersionByID(ctx, details.TableDescs[i].ID, txn)
+		if err != nil {
+			return err
+		}
+		// Ensure that the version matches we expect. In the case that it doesn't,
```

Comment:
IIUC the descriptors could either still be offline or be public at this point. If they're already public, we can't actually make any guarantees about their versions, right?